lib/server.js

```diff
@@ -119,8 +119,8 @@ class Server extends EventEmitter {
       var opt = Object.assign({}, clone(self.clientOptions), options);
       opt.host = self.options.bind_ip;
       opt.port = self.options.port;
-      opt.connectionTimeout = 5000;
-      opt.socketTimeout = 5000;
+      opt.connectionTimeout = options.connectionTimeout || 5000;
```

Comment:
Is there a further refactor we can do to deduplicate the number of places we do this same "basic options resolution"?

Reply:
Not that I can think of. Besides saving an instance to `Server`, we will have to include this each time.

Reply:
I was thinking something like a common function such as:
```
function cloneClientOptions(manager, options) {
  const result = Object.assign({}, clone(manager.clientOptions), options);
  options.host = self.options.bind_ip;
  options.port = self.options.port;
  options.connectionTimeout = typeof options.connectionTimeout === 'number' ? options.connectionTimeout : 5000;
  options.socketTimeout = typeof options.socketTimeout === 'number' ? options.socketTimeout : 0;

  return options;
}
```

These options are cloned off every time we create an instance of the mongo client with the same default checking. My thought was if we refactor to use a single method for this, you won't have to remember to change it in 6 locations if something changes in the future.

Reply:
Oh sure! I misunderstood what you meant by refactor. I was thinking you meant store the socket & connection timeout parameters within the `CoreServer` instance of mongodb-core.

I think we're at a fork in the road. If you see my comment below, I think we should do one of two things:
1. Place the two parameters in `self.clientOptions` and if we do that. The default values of 5000 & 0 will be handled in the constructor just once. Thus, I don't think the `cloneClientOptions` function will be necessary.
2. Keep the two parameters in each function. Then I definitely like your function (thinking in utils.js and calling from Mongos & Server classes).

Just need to know what you prefer, I'm fine either way. I think I'm leaning towards option 1 more now.

Reply:
I added what we discussed. Made a few changes I wanted to describe to you in case you were wondering:
* Renamed to `createCommandOptions` because there is the bind_ip & port being set which is specific to the `CoreServer` class. Didn't want to give people the idea that the function **just** cloned options.
* Removed `localOptions` from being assigned to the options. Therefore, the only local options that will be copied over after the connection/socket timeouts.
    * I justified this because if we copied the local options, then there would be socketTimeoutMS & connectionTimeoutMS options but they would not mean anything because the options are socketTimeout & connectionTimeout in `CoreServer`.

Reply:
SGTM, resolving. Will bring up any issues with implementation during re-review